main.py

```diff
@@ -44,13 +44,16 @@ async def lifespan(_app: FastAPI):
 async def list_directory(directory: str = "") -> List[str]:
     """Gets the list of elements in the given path and returns it.
 
-    The "master_path" in the configuration file is used for the prefix of the path.
+    The "master_path" and "repository_path" in the configuration file 
+    is used for the prefix of the path.
 
     Args:
         directory: The path of the directory to search for.
     """
     remote = get_client("master_experiment_db")
-    return remote.list_directory(posixpath.join(configs["master_path"], directory))
+    return remote.list_directory(posixpath.join(
+        configs["master_path"], configs["repository_path"], directory
```

Comment:
Is the repository path always in the master path?
If not, I think we don't have to put them together.
What would you prefer: absolute vs. relative path for the repository path?

Reply:
I don't know if the repository path must be in the master path, but the repository passed as the `repository` argument of `artiq_master` is defined as a relative path to the master path!
I inteded that the server manager set `repository_path` the same as the argument.

Please refer to [artiq_master](https://m-labs.hk/artiq/manual/management_system.html#repository).